test_rmw_implementation/test/test_subscription.cpp

```diff
@@ -377,3 +377,53 @@ TEST_F(CLASSNAME(TestSubscriptionUse, RMW_IMPLEMENTATION), count_mismatched_subs
   EXPECT_EQ(RMW_RET_OK, ret) << rmw_get_error_string().str;
   EXPECT_EQ(0u, publisher_count);
 }
+
+TEST_F(CLASSNAME(TestSubscriptionUse, RMW_IMPLEMENTATION), take_serialized_with_bad_args) {
+  rmw_subscription_allocation_t * null_allocation{nullptr};  // still valid allocation
+  rcutils_allocator_t default_allocator = rcutils_get_default_allocator();
+  bool taken = false;
+  rmw_serialized_message_t serialized_message = rmw_get_zero_initialized_serialized_message();
+  rmw_serialized_message_t original_message = serialized_message;
+  ASSERT_EQ(
+    RMW_RET_OK, rmw_serialized_message_init(
+      &serialized_message, 1lu, &default_allocator)) << rmw_get_error_string().str;
+  ASSERT_EQ(
```

Comment:
If this asserts fails and exits the test early, the rmw_serialized_message_fini is never called for the first msg. You'll have to use a `OSRF_SCOPE_EXIT` macro to make the fini calls in this scenario

Reply:
Could you point me to some example of that?

Reply:
Updated on b1fca8c